### PR TITLE
Documentation: edits to main page

### PR DESCRIPTION
**Pull request by mattwelborn**
## Description
This PR makes edits to the front page of the documentation. The primary change is the replacement of the OpenFF workflow graphic with an adaptation of @Lnaden 's pipeline slide.

## Status
- [ ] Changelog updated
- [ ] Ready to go

### CODE REVIEW

File: docs/qcfractal/source/index.rst [hunk 1]
```diff
@@ -18,18 +18,32 @@ QCFractal emphasizes the following virtues:
...
 
+QCFratal within the QCArchive stack
+-----------------------------------
+
+Within the QCArchive stack, QCFractal is responsible for:
+
+- Accepting requests for calculations, submitted from a client such as `QCPortal <http://docs.qcarchive.molssi.org/projects/qcportal/en/latest/>`_. 
+- Running these calculations on disparate compute resources through Quantum Chemistry programs and procedures supported by `QCEngine <http://docs.qcarchive.molssi.org/projects/qcengine/en/latest/>`_. 
+- Storing and organizing the results of these calculations in a database.
+- Providing access to stored results to clients such as QCPortal.
```
Comment (dgasmith) [inline docs/qcfractal/source/index.rst]: Link to portal again?

File: docs/qcfractal/source/index.rst [hunk 2]
```diff
@@ -18,18 +18,32 @@ QCFractal emphasizes the following virtues:
...
+
+The diagram below illustrates a representative data flow:
+
+.. image:: media/QCEcosystem.png
+   :width: 800px
+   :alt: QCArchive ecosystem
+   :align: center
```
Comment (dgasmith) [inline docs/qcfractal/source/index.rst]: The image likely needs an update. The party parrot as mentioned should probably be replaced, QCPortal needs a font similar to the other programs, and "compute through engine" reads awkwardly in the image. Did you obtain the source image from @Lnaden?

File: docs/qcfractal/source/index.rst (hunk 1 above)
Comment (mattwelborn) [reply docs/qcfractal/source/index.rst]: done.

File: docs/qcfractal/source/index.rst (hunk 2 above)
Comment (mattwelborn) [reply docs/qcfractal/source/index.rst]: @dgasmith thanks for the feedback!

**Comment by dgasmith** [comment]
@mattwelborn Is this ready to go?

**Comment by mattwelborn** [comment]
Yes.

**Pull request merged**